ugs-core/src/com/willwinder/universalgcodesender/model/PartialPosition.java

```diff
@@ -0,0 +1,149 @@
+package com.willwinder.universalgcodesender.model;
+
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a maybe partial coordinate (ie. only certain axis-values are known, eg. for moves where only some axis
+ * are changed)
+ */
+public class PartialPosition {
```

Comment:
I like this, nice work. 

Did you consider using Optionals which would allow us to scrap the has-methods. The get-methods would just return a `Optional.ofNullable(x)`.
I don't know if it gets any cleaner though.  To use one could do:
```
partialPosition.getX()
    .ifPresent(x -> sb.append("X").append(Utils.formatter.format(x)).append(" "));
```

Reply:
The idea was that `PartialPosition` handles the String formatting... but for some reasons I put them into `GcodeUtils.formatPartialPosition(xyPosition)` -> ive now moved it into `PartialPosition` to make this more clear. Then it shouldnt be necessary (at least not that often) to individually check on `hasX()`...